client/layout/transient-notices/index.js

```diff
@@ -23,6 +35,30 @@ function TransientNotices( props ) {
 		return { notices, notices2 };
 	} );
 
+	const dequeueNotice = ( id ) => {
+		const remainingNotices = queueNotices.filter(
+			( notice ) => notice.id !== id
+		);
+		setQueuedNotices( remainingNotices );
+		updateOptions( {
+			[ QUEUE_OPTION ]: remainingNotices,
+		} );
+	};
+
+	useEffect( () => {
+		queueNotices.forEach( ( queuedNotice ) => {
+			const notice = applyFilters( QUEUED_NOTICE_FILTER, {
+				queuedNotice,
+			} );
+			createNotice( notice.status, notice.content, {
```

Comment:
I think this may have been in an older version of this branch.  This needed to be updated to:

```
const notice = applyFilters( QUEUED_NOTICE_FILTER, queuedNotice );
```

Reply:
@joshuatf sorry I'm coming in a little naive here, so I'll start with a question,

is it desired to support adding transient notices via the client side as well as the backend? or is this purely a "per page load" feature.

I'm just asking that because the way this useEffect works, you won't be able to get any new transient notices if they were somehow added via the client, it will show them on first mount, but any further changes won't be reflected. Just checking really what the desired behaviour is.

Reply:
Ping @joshuatf. I don't _think_ this would be an issue as one could just call `useDispatch( 'core/notices2' ).createNotice()` directly to create a notice, unless I'm completely misunderstanding things.

Reply:
Yes I think thats correct. The feature describes itself as "Transient notices provide a way to display snackbar notices from the server." so this seems entirely a feature for displaying a notice from a single server page load only. Of course creating notices via another mechanism would still work. 

I guess its worth mentioning that this wouldn't work in a case where you wanted to raise a notice during an API call (ie we fetch some data and there is no page load) and expect it to be displayed on screen. I think thats probably an edge case and I can't imagine that needing to be done, but just calling it out anyway.

One thing thats kinda weird here though is how `queuedNotices` is a piece of state that can change. Whats the point of storing it as state and updating it in dequeue? It makes sense to update the option but maintaining state from a single page load that can't change with this `useEffect` being basically the equivalent of onMount means that you'll never care about whats in queuedNotices after the initial page load?